core/shared/src/main/scala/cats/parse/Parser.scala

```diff
@@ -1722,6 +1731,22 @@ object Parser extends ParserInstances {
       override def parseMut(state: State): B = Impl.map(parser, fn, state)
     }
 
+    final def select[A, B](
+        parser: Parser[Either[A, B]],
+        fn: Parser[A => B],
+        state: State
+    ): B = {
+      val either = parser.parseMut(state)
```

Comment:
Actually, look at flatMap. I think you have to set capture to true here unconditionally since otherwise we don't know what to do next.

After we capture the first, we can set capture to false on the second part since we can just avoid applying the function.